pkg/codegen/utilities.go

```diff
@@ -113,16 +112,13 @@ func (s Set) Has(v interface{}) bool {
 	return ok
 }
 
-// SortedKeys returns a sorted list of keys for the given map. The map's key type must be of kind string.
-func SortedKeys(m interface{}) []string {
-	mv := reflect.ValueOf(m)
-
-	contract.Require(mv.Type().Kind() == reflect.Map, "m")
-	contract.Require(mv.Type().Key().Kind() == reflect.String, "m")
-
-	keys := make([]string, mv.Len())
-	for i, k := range mv.MapKeys() {
-		keys[i] = k.String()
+// SortedKeys returns a sorted list of keys for the given map.
+func SortedKeys[T any](m map[string]T) []string {
+	keys := make([]string, len(m))
+	var i int
```

Comment:
Smallest of nits:
Rather than track an index, this might be better done with zero length and `append`.

```go
keys := make([]string, 0, len(m))
for k := range m {
  keys = append(keys, k)
}
```

(Assuming we keep this implementation and don't use `x/exp/{slices,maps}`.)